Document the fact that entity fragments exists (#1115)
RoadieHQ/marketing-site: content/docs/details/decorating-components/index.md

---
title: Introduction
publishedDate: '2023-09-05'
description: An introduction about decorating components.
---

> Catalog is built around the concept of [entity descriptor YAML files](https://backstage.io/docs/features/software-catalog/descriptor-format/), therefore, any change you wish to make to your components, needs to be done via editing those files. 

Reviewer: This isn't quite true. The catalog doesn't depend on YAML files but can be populated using any methods

> However, as YAML files can grow in complexity, they can become challenging to read and maintain. Editing those files, can sometimes be a long process, especially if ownership is shared accross the teams in your organization.

Reviewer: It's likely better to take an approach where we talk about multiple sources of data.

![Decorate component](./decorate_entity.png)

This will navigate you to the page where you can decorate your component with links and annotations, without a need to manually edit YAML files in your version control system. All of this is done via UI, in specifically designated sections.

A visual representation of the YAML structure of the component, is shown in Existing Entity section of the page.

![Existing component](./existing_entity.png)

Adding links is done via 'Add link' button in 'Links' section. 

![Add link](./add_link.png)

New link properties need to follow [link properties types](https://backstage.io/docs/features/software-catalog/descriptor-format#links-optional). Usually semantics of the type field are undefined, but we have added few predefined options you can select from a dropdown list.

Adding annotations is done via 'Add annotation' button in 'Annotations' section.

![Add annotation](./add_annotation.png)

New annotation properties need to follow [annotation properties types](https://backstage.io/docs/features/software-catalog/descriptor-format#annotations-optional). 

> You will find all existing annotations used in components accross catalog in a dropdown list.

Reviewer: ```suggestion
You will find all existing annotations used in components across the catalog in a dropdown list.
```

> When you are happy with added annotations or/and links simply click 'Save' button and you will shortly see you changes.

Reviewer: ```suggestion
When you are happy with added annotations and/or links simply click 'Save' button and you will shortly see you changes.
```

---
title: Introduction
publishedDate: '2023-09-05'
description: An introduction about decorating components.
---

> Components within the software catalog are structured around [metadata YAML files](https://backstage.io/docs/features/software-catalog/descriptor-format/) alongside the code, which are then collected and presented in Backstage for visualization. The source of truth for the components are mentioned YAML files, stored in source control such as GitHub, GitHub Enterprise, GitLab, Bitbucket, and others. 

Reviewer 2: ```suggestion
Decorators are an easy way to add links and annotations to Entities, without editing the YAML file that the Entity originates from.

Decorators are stored inside Roadie and available via the API.
```

> Because the process of adding components to the catalog can be achieved through manual registration, creating new components within Backstage, or integrating with an external source, the catalog may often comprise components with different sources of data. Consequently, modifying these components can be challenging, as it involves making changes in multiple locations.

Reviewer 2: ```suggestion
```

We don't need to explain why the feature exists in technical documentation.

> Idea behind component decorator is to provide an easy way to modify components in one place, through Backstage, and enhance usability for components used accross organization.

Reviewer 2: ```suggestion
```
